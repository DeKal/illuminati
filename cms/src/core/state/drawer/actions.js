import { createAction } from 'redux-actions'
import { OPEN_DRAWER, CLOSE_DRAWER } from 'core/const/actionType'

export const openDrawer = createAction(OPEN_DRAWER)
export const closeDrawer = createAction(CLOSE_DRAWER)
