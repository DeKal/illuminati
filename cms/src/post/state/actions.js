import { createAction } from 'redux-actions'
import {
  FETCH_POST,
  EDIT_POST_TITLE,
  EDIT_POST_SLUG,
  EDIT_POST_BRIEF,
  EDIT_POST_AUTHOR,
  EDIT_POST_TIME,
  EDIT_POST_TAGS,
  EDIT_POST_CONTENT,
  SAVE_EDITED_CONTENT
} from 'core/state/actionType'
import { toRequest } from 'core/state/utils'

export const fetchPost = createAction(toRequest(FETCH_POST))
export const editPostTitle = createAction(EDIT_POST_TITLE)
export const editPostSlug = createAction(EDIT_POST_SLUG)
export const editPostBrief = createAction(EDIT_POST_BRIEF)
export const editPostAuthor = createAction(EDIT_POST_AUTHOR)
export const editPostTime = createAction(EDIT_POST_TIME)
export const editPostTags = createAction(EDIT_POST_TAGS)
export const editPostContent = createAction(EDIT_POST_CONTENT)
export const saveEditedContent = createAction(toRequest(SAVE_EDITED_CONTENT))
