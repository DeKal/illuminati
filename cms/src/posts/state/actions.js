import { createAction } from 'redux-actions'
import {
  CHANGE_PAGE,
  CHANGE_ROW_PER_PAGE,
  SET_ORDER,
  SET_ORDER_COLUMN,
  SET_SELECTED_POST,
  SELECT_ALL_POSTS,
  CLEAR_SELECTED
} from 'core/const/actionType'

export const changePage = createAction(CHANGE_PAGE)

export const changeRowsPerPage = createAction(CHANGE_ROW_PER_PAGE)

export const setOrder = createAction(SET_ORDER)

export const setOrderColumn = createAction(SET_ORDER_COLUMN)

export const setSelectedPost = createAction(SET_SELECTED_POST)

export const selectAllPosts = createAction(SELECT_ALL_POSTS)

export const clearSelected = createAction(CLEAR_SELECTED)
