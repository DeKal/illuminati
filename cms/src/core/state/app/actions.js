import { createAction } from 'redux-actions'
import { ERROR } from 'core/state/actionType'

export const assignError = createAction(ERROR)
