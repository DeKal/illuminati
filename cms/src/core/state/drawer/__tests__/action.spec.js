import { openDrawer, closeDrawer } from '../actions'

describe('drawer actions', () => {
  it('should return action with open drawer', function() {
    expect(openDrawer()).toMatchSnapshot()
  })

  it('should return action with close drawer', function() {
    expect(closeDrawer()).toMatchSnapshot()
  })
})
