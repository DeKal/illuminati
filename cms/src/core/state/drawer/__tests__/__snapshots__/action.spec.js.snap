// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`drawer actions should return action with close drawer 1`] = `
Object {
  "type": "drawer/closeDrawer",
}
`;

exports[`drawer actions should return action with open drawer 1`] = `
Object {
  "type": "drawer/openDrawer",
}
`;
